brain-marks/Services/MigrationService.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  MigrationService.swift
+//  brain-marks
+//
+//  Created by Jay on 12/13/22.
+//
+
+import CoreData
+import Foundation
+
+/// Controls the migration from Amplify to CoreData
+class MigrationService {
+    private let storageProvider = StorageProvider.shared
+    private var awsCategories: [AWSCategory] = []
+
+    private let amplifyDataStore = DataStoreManger.shared
+
+    private var managedObjectContext: NSManagedObjectContext
+
+    init() {
+        self.managedObjectContext = storageProvider.context
+    }
+
+    func performMigration() {
+        /// 1. Fetch categories
+        getCategories()
```

Comment:
technically because `getCategories` is called here and sets the variable `awsCategories` is it possible for the function to still be fetching data, before moving onto the next step? (iterating through `awsCategories`)

I want to say maybe? and because of this, let's not use this function `getCategories` and instead use it directly in `performMigration` even if the code isn't that clean. This also makes sure that the UserDefault bool doesn't get set as true, even if this `getCategories` failed for some reason

Reply:
Am I understanding this right, move the `amplifyDataStore.fetchCategories` into `performMigration`?

I think that makes a bit more sense also